app/models/event.rb

```diff
@@ -0,0 +1,57 @@
+# == Schema Information
+#
+# Table name: events
+#
+#  id              :bigint           not null, primary key
+#  data            :jsonb
+#  handleable_type :string
+#  name            :string
+#  type            :string
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  handleable_id   :bigint
+#
+# Indexes
+#
+#  index_events_on_handleable_type_and_handleable_id  (handleable_type,handleable_id)
+#
+
+class Event < ApplicationRecord
+  belongs_to :handleable, polymorphic: true, required: false
+  validates :name, :data, presence: true
+
+  EVENTS = %w[pull_request].freeze
+  private_constant :EVENTS
+
+  def resolve
+    return handle if handleable?
+
+    save!
+  end
+
+  private
+
+  def handle
+    Event.create!(build_attributes)
+  end
+
+  def build_attributes
+    {
+      handleable: resolve_event,
```

Comment:
Are we creating the `handleable` object here? I'm asking because the `opened` method in the GithubService already creates the PullRequest object.

Reply:
No, it will depend on the `action` that we are receiving, if the event has the `opened` action it will create a PR, otherwise it will find the PR and assign it there.